osquery/logger/plugins/http_logger.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant 
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <exception>
+#include <mutex>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/flags.h>
+#include <osquery/logger.h>
+#include <osquery/registry.h>
+#include <osquery/database.h>
+
+#define LOG_KEY "http_logger"
+
+namespace osquery {
+
+FLAG(string,
+     logger_uri,
+     "Not set",
```

Comment:
Maybe set this default to a blank string, then a `std::string::size` can decide if you bail before making a request?